### PR TITLE
feat: allow Laravel 10

### PR DESCRIPTION
**Pull request by 77web**
Now that Laravel 10 has been released, we need Laravel 10 compatible version of laravel-doctrine packages :)
orm and extensions have already PRs, so I want to try Laravel 10 along with migrations.

Please review and try, friends.
Thank you.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,14 +17,20 @@
...
     "doctrine/migrations": "^3.4",
     "doctrine/dbal": "^2.10.1|^3",
-    "illuminate/config": "^6.0|^7.0|^8.0|^9.0",
-    "illuminate/contracts": "^6.0|^7.0|^8.0|^9.0",
-    "illuminate/console": "^6.0|^7.0|^8.0|^9.0",
-    "laravel-doctrine/orm": "^1 | ^2@dev"
+    "illuminate/config": "^6.0|^7.0|^8.0|^9.0|^10.0",
+    "illuminate/contracts": "^6.0|^7.0|^8.0|^9.0|^10.0",
+    "illuminate/console": "^6.0|^7.0|^8.0|^9.0|^10.0",
+    "laravel-doctrine/orm": "dev-l10-compatibility"
```
Comment (eigan) [inline composer.json]: That branch will most likely not work as it upgraded to DBAL 3 while not considering all the breaking changes. A PR addressed DBAL 3, and we are now focusing on fixing remaining deprecations before L10. 
You could open a PR for L10 on the 2.x branch in orm though :) 

See https://github.com/laravel-doctrine/orm/issues/542

Comment (77web) [reply composer.json]: @eigan Thank you! I've opened a PR in laravel-doctrine/orm, and changed the branch name!

File: composer.json [hunk 2]
```diff
@@ -17,14 +17,20 @@
...
+    }
+  ],
   "require": {
     "php": "^7.4|^8",
     "doctrine/migrations": "^3.4",
     "doctrine/dbal": "^2.10.1|^3",
-    "illuminate/config": "^6.0|^7.0|^8.0|^9.0",
-    "illuminate/contracts": "^6.0|^7.0|^8.0|^9.0",
-    "illuminate/console": "^6.0|^7.0|^8.0|^9.0",
-    "laravel-doctrine/orm": "^1 | ^2@dev"
```
Comment (eigan) [inline composer.json:27]: You can revert this change now (and remove the `repositories` part) :)

Comment (eigan) [reply composer.json:27]: Same with the PR in extensions repo

Comment (77web) [reply composer.json:27]: done 😄

Comment (eigan) [reply composer.json:27]: Should still allow 1^, so `^1 | ^2@dev`  is correct here :)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       matrix:
         php: ['8.0', '8.1', '8.2']
-        laravel: ['8', '9']
+        laravel: ['8', '9', '10']
         exclude:
           - php: '8.2'
```
Comment (eigan) [inline .github/workflows/ci.yml:16]: Add another exclude here

```
          - php: '8.0'
            laravel: '10'
```

Comment (77web) [reply .github/workflows/ci.yml:16]: Thanks, fixed!
